cache/cache_entry_roles.h

```diff
@@ -0,0 +1,122 @@
+//  Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved.
+//  This source code is licensed under both the GPLv2 (found in the
+//  COPYING file in the root directory) and Apache 2.0 License
+//  (found in the LICENSE.Apache file in the root directory).
+
+#pragma once
+
+#include <array>
+#include <cstdint>
+#include <unordered_map>
+
+#include "rocksdb/cache.h"
+
+namespace ROCKSDB_NAMESPACE {
+
+// Classifications of block cache entries, for reporting statistics
+enum class CacheEntryRole {
+  // Block-based table data block
+  kDataBlock,
+  // Block-based table filter block (full or partitioned)
+  kFilterBlock,
+  // Block-based table metadata block for partitioned filter
+  kFilterMetaBlock,
```

Comment:
Yes, this is for the top-level index of filter partitions. I don't think it needs to be split out for indexes because top-level and partitions use the same format, so you don't need it to answer much space is used by blocks of a particular schema. And you can already answer the question of how big the top-level index size is with table properties. Whether or not it's in cache is predictable.

In a future with hash-partitioned filters, there would likely be a different kind of filter metadata/index block. I don't care so much about distinguishing which kind of filter metadata it is as I do about keeping the accounting separate from the underlying filters. I chose `kFilterMetaBlock` so an existing name would be appropriate in that future case.

I'm not married to the categories I put here, so if you have more thoughts, share! And hopefully I've kept the door open to change.

Reply:
I also think it would take some extra code to draw out the top-level index or not distinction from the context where the deleter is chosen. For filters, it's easy because it's a different `TBlocklike`